src/knightos/color.c

```diff
@@ -0,0 +1,87 @@
+#include <knightos/color.h>
+#include <kernel.h>
+#include <stdbool.h>
+
+bool is_legacy() {
+	__asm
+	PCALL(CHECKLEGACYLCDMODE)
+	LD L, 0
+	JR Z, .legacy
+.legacy:
+	INC L
+	__endasm;
+}
+
+bool color_supported() {
+	__asm
+	PCALL(COLORSUPPORTED)
+	LD L, 0
+	JR NZ, .notSupported
+	INC L
+.notSupported:
+	LD (_errno), A
+	__endasm;
+}
+
+void reset_legacy_lcd_mode() {
+	__asm
+	PCALL(RESETLEGACYLCDMODE)
+	__endasm;
+}
+
+void clear_color_lcd(char *color) {
+	__asm
+	POP IY ; color
```

Comment:
Also, the type of this should probably be uint16_t, not char *. It's not a char *.